board/prepare.init

```diff
@@ -251,74 +205,101 @@ wb_prepare_filesystems()
+        log_action_begin_msg "Creating ${descr}"
 
-        if [[ -e "/mnt/data/${eth0_mac}" ]]; then
-            cp "/mnt/data/${eth0_mac}" "${eth0_mac}"
-        elif [[ -e "$SERIAL" ]]; then
-            cp "$SERIAL" "${eth0_mac}"
-        elif [[ -e "/mnt/data/$SERIAL" ]]; then
-            cp "/mnt/data/$SERIAL" "${eth0_mac}"
+        if [[ -e "/mnt/data/${file}" ]]; then
+            log_action_cont_msg "found at /mnt/data"
+            cp "/mnt/data/${file}" "${file}"
         else
-            wb-gen-serial -m 0 > ${eth0_mac}
+            $cmd ${@:4} > $1
         fi
 
         log_end_msg $?
     fi
+}
 
-    if [[ ! -f "${eth1_mac}" ]]; then
-        log_action_begin_msg "Getting eth1 MAC address"
-
-        if [[ -e "/mnt/data/${eth1_mac}" ]]; then
-            cp "/mnt/data/${eth1_mac}" "${eth1_mac}"
+wb_gen_mac()
+{
+    local ethid=$1
+
+    if [[ "${ethid}" == "0" ]]; then
+        if [[ -e "${SERIAL}" ]]; then
+            log_action_cont_msg "found at ${SERIAL}" >&2
+            cat ${SERIAL}
+        elif [[ -e "/mnt/data/${SERIAL}" ]]; then
+            log_action_cont_msg "found at /mnt/data/${SERIAL}" >&2
+            cat "/mnt/data/${SERIAL}"
         else
-            wb-gen-serial -m 1 > ${eth1_mac}
+            wb-gen-serial -m ${ethid}
         fi
-
-        log_end_msg $?
+    else
+        wb-gen-serial -m ${ethid}
     fi
+}
+
+wb_fix_macs()
+{
+    local eth0_mac="$WB_DIR/eth0_mac.conf"
+    local eth1_mac="$WB_DIR/eth1_mac.conf"
+
+    wb_fix_file ${eth0_mac} "eth0 MAC address" wb_gen_mac 0
+    wb_fix_file ${eth1_mac} "eth1 MAC address" wb_gen_mac 1
 
     return 0
 }
 
+wb_fix_serial()
+{
+    wb_fix_file ${SERIAL} "old eth0 MAC aka serial.conf" wb-gen-serial -m
+}
+
 wb_fix_short_sn()
 {
-    [[ ! -f "$SHORT_SN_FNAME" ]] && {
-        log_action_msg "Storing short SN"
-        [[ -e "/mnt/data/$SHORT_SN_FNAME" ]] &&
-            cp "/mnt/data/$SHORT_SN_FNAME" "$SHORT_SN_FNAME" || {
-            local short_sn=`wb-gen-serial -s`
-            echo "$short_sn" > "$SHORT_SN_FNAME"
-        }
-    }
+    wb_fix_file ${SHORT_SN_FNAME} "short serial number" wb-gen-serial -s
+
+    # also fill everything containing this short serial number
+    local short_sn=`cat ${SHORT_SN_FNAME}`
+    local ssid="WirenBoard-${short_sn}"
+    local hostname="wirenboard-${short_sn}"
+
+    log_action_msg "Creating hostname file with ${hostname}"
+    echo $hostname > /etc/hostname.wb
+    log_end_msg $?
+
+    log_action_msg "Setting internal Wi-Fi SSID to ${ssid}"
+    sed -i.orig "s/^ssid=.*/ssid=${ssid}/" "/etc/hostapd.conf.wb"
+    log_end_msg $?
+}
+
+# returns 0 if this boot is first and 1 if it is not
+wb_is_firstboot()
+{
+    [[ ! -f $FIRSTBOOT_FLAG ]]
 }
 
 # This function should be called only on first boot of the rootfs
 wb_firstboot()
 {
-    local flag="/var/lib/firstboot_done.flag"
-
-    [[ -f $flag ]] && return 0
-
     log_action_msg "Preparing rootfs for the first boot"
     service watchdog stop
```

Comment:
Do someone remember why we need this?

Reply:
yes, it indicates if the script has been successfully executed or not. At first run it sets partitions, etc, but there is no need for this later